day5/day5.c

```diff
@@ -0,0 +1,94 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define YES 1
+#define NO 0
+
+struct rule {
+	int value;
+	struct rule *next;
+};
+void insert (struct rule *list[], int idx, int value)
+{
+	struct rule *node = malloc(sizeof(struct rule));
+	if (node == NULL)
+		return;
+	node->value = value;
+	node->next = list[idx];
+	list[idx] = node;
+}
+void clear_list(struct rule *list[])
+{
+	int idx;
+	for (idx = 0; idx < 100; idx++) {
+		struct rule *p = list[idx];
+		while (p != NULL) {
+			struct rule *tmp = p;
+			p = p->next;
+			free(tmp);
+		}
+		list[idx] = NULL;
+	}
+}
+int main(void)
+{
+	FILE *f;
+	struct rule *list[100], *print;
+	int i, j, pages[25], size, idx, value, correct, current, cp;
+	int middles = 0;
+	int mcorrected = 0;
```

Comment:
"zero-initialization" is something interesting to know: what are the default values for different types?
When you first declare an `int` value it's already initialized as 0, so you don really need to explicitly set it.
Every type has a default "zero-initialization" value. Check it for future cases ;)